<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8" />
    <meta name="robots" content="index, follow, all" />
    <title>PHPUnit\Framework\ExceptionWrapper | Aventri Connect API</title>

            <link rel="stylesheet" type="text/css" href="../../css/bootstrap.min.css">
        <link rel="stylesheet" type="text/css" href="../../css/bootstrap-theme.min.css">
        <link rel="stylesheet" type="text/css" href="../../css/doctum.css">
        <link rel="stylesheet" type="text/css" href="../../fonts/doctum-font.css">
        <script src="../../js/jquery-3.5.1.slim.min.js"></script>
        <script async defer src="../../js/bootstrap.min.js"></script>
        <script src="../../js/typeahead.min.js"></script>
        <script src="../../doctum.js"></script>
        <meta name="MobileOptimized" content="width">
        <meta name="HandheldFriendly" content="true">
        <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1">
    </head>

    <body id="class" data-name="class:PHPUnit_Framework_ExceptionWrapper" data-root-path="../../">
            <div id="content">
        <div id="left-column">
                <div id="control-panel">
                <form id="search-form" action="../../search.html">
            <span class="icon icon-search"></span>
            <input name="search"
                   class="typeahead form-control"
                   type="search"
                   placeholder="Search">
        </form>
    </div>

                <div id="api-tree"></div>

        </div>
        <div id="right-column">
                <nav id="site-nav" class="navbar navbar-default" role="navigation">
        <div class="container-fluid">
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar-elements">
                    <span class="sr-only">Toggle navigation</span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="../../index.html">Aventri Connect API</a>
            </div>
            <div class="collapse navbar-collapse" id="navbar-elements">
                <ul class="nav navbar-nav">
                    <li><a href="../../classes.html">Classes</a></li>
                    <li><a href="../../namespaces.html">Namespaces</a></li>
                    <li><a href="../../interfaces.html">Interfaces</a></li>
                    <li><a href="../../traits.html">Traits</a></li>
                    <li><a href="../../doc-index.html">Index</a></li>
                    <li><a href="../../search.html">Search</a></li>
                </ul>
            </div>
        </div>
    </nav>

                        <div class="namespace-breadcrumbs">
            <ol class="breadcrumb">
                <li><span class="label label-default">class</span></li>
                        <li><a href="../../PHPUnit.html">PHPUnit</a></li><li class="backslash">\</li><li><a href="../../PHPUnit/Framework.html">Framework</a></li><li class="backslash">\</li><li>ExceptionWrapper</li>
            </ol>
        </div>
                <div id="page-content">
                
    <div class="page-header">
        <h1>
            ExceptionWrapper
                
        </h1>
    </div>

    <p>final         class
    <strong>ExceptionWrapper</strong>        extends <a href="../../PHPUnit/Framework/Exception.html"><abbr title="PHPUnit\Framework\Exception">Exception</abbr></a>
</p>

        
    

            <div class="description">
            <p>Wraps Exceptions thrown by code under test.</p>            <p>Re-instantiates Exceptions thrown by user-space code to retain their original
class names, properties, and stack traces (but without arguments).</p>

<p>Unlike PHPUnit\Framework\Exception, the complete stack of previous Exceptions
is processed.</p>                    </div>
    
    
    
            <h2>Properties</h2>

            <table class="table table-condensed">
                    <tr>
                <td class="type" id="property_serializableTrace">
                                        protected                                        array
                </td>
                <td>$serializableTrace</td>
                <td class="last"></td>
                <td><small>from&nbsp;<a href="../../PHPUnit/Framework/Exception.html#property_serializableTrace">
<abbr title="PHPUnit\Framework\Exception">Exception</abbr></a></small></td>
            </tr>
                    <tr>
                <td class="type" id="property_className">
                                        protected                                        string
                </td>
                <td>$className</td>
                <td class="last"></td>
                <td></td>
            </tr>
                    <tr>
                <td class="type" id="property_previous">
                                        protected                                        null|ExceptionWrapper
                </td>
                <td>$previous</td>
                <td class="last"></td>
                <td></td>
            </tr>
            </table>

    
            <h2>Methods</h2>

            <div class="container-fluid underlined">
                    <div class="row">
                <div class="col-md-2 type">
                    
                </div>
                <div class="col-md-8 type">
                    <a href="#method___construct">__construct</a>(<a target="_blank" rel="noopener" href="https://www.php.net/Throwable">Throwable</a> $t)
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    string
                </div>
                <div class="col-md-8 type">
                    <a href="#method___toString">__toString</a>()
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    array
                </div>
                <div class="col-md-8 type">
                    <a href="#method___sleep">__sleep</a>()
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"><small>from&nbsp;<a href="../../PHPUnit/Framework/Exception.html#method___sleep">
<abbr title="PHPUnit\Framework\Exception">Exception</abbr></a></small></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    array
                </div>
                <div class="col-md-8 type">
                    <a href="#method_getSerializableTrace">getSerializableTrace</a>()
        
                                            <p>Returns the serializable trace (without 'args').</p>                </div>
                <div class="col-md-2"><small>from&nbsp;<a href="../../PHPUnit/Framework/Exception.html#method_getSerializableTrace">
<abbr title="PHPUnit\Framework\Exception">Exception</abbr></a></small></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    string
                </div>
                <div class="col-md-8 type">
                    <a href="#method_getClassName">getClassName</a>()
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    <a href="../../PHPUnit/Framework/ExceptionWrapper.html"><abbr title="PHPUnit\Framework\ExceptionWrapper">ExceptionWrapper</abbr></a>|null
                </div>
                <div class="col-md-8 type">
                    <a href="#method_getPreviousWrapped">getPreviousWrapped</a>()
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    void
                </div>
                <div class="col-md-8 type">
                    <a href="#method_setClassName">setClassName</a>(string $className)
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    void
                </div>
                <div class="col-md-8 type">
                    <a href="#method_setOriginalException">setOriginalException</a>(<a target="_blank" rel="noopener" href="https://www.php.net/Throwable">Throwable</a> $t)
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"></div>
            </div>
                    <div class="row">
                <div class="col-md-2 type">
                    <abbr title="PHPUnit\Framework\Throwable">Throwable</abbr>|null
                </div>
                <div class="col-md-8 type">
                    <a href="#method_getOriginalException">getOriginalException</a>()
        
                                            <p class="no-description">No description</p>
                                    </div>
                <div class="col-md-2"></div>
            </div>
            </div>


        <h2>Details</h2>

            <div id="method-details">
                    <div class="method-item">
                    <h3 id="method___construct">
        <div class="location">at line 41</div>
        <code>                    
    <strong>__construct</strong>(<a target="_blank" rel="noopener" href="https://www.php.net/Throwable">Throwable</a> $t)
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
                            <h4>Parameters</h4>

                    <table class="table table-condensed">
                    <tr>
                <td><a target="_blank" rel="noopener" href="https://www.php.net/Throwable">Throwable</a></td>
                <td>$t</td>
                <td></td>
            </tr>
            </table>

            
            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method___toString">
        <div class="location">at line 50</div>
        <code>                    string
    <strong>__toString</strong>()
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td>string</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method___sleep">
        <div class="location">in <a href="../../PHPUnit/Framework/Exception.html#method___sleep">
<abbr title="PHPUnit\Framework\Exception">Exception</abbr></a> at line 69</div>
        <code>                    array
    <strong>__sleep</strong>()
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td>array</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method_getSerializableTrace">
        <div class="location">in <a href="../../PHPUnit/Framework/Exception.html#method_getSerializableTrace">
<abbr title="PHPUnit\Framework\Exception">Exception</abbr></a> at line 77</div>
        <code>                    array
    <strong>getSerializableTrace</strong>()
        </code>
    </h3>
    <div class="details">
            
    

                    <div class="method-description">
                                    <p>Returns the serializable trace (without 'args').</p>                                                </div>
                <div class="tags">
            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td>array</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method_getClassName">
        <div class="location">at line 65</div>
        <code>                    string
    <strong>getClassName</strong>()
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td>string</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method_getPreviousWrapped">
        <div class="location">at line 70</div>
        <code>                    <a href="../../PHPUnit/Framework/ExceptionWrapper.html"><abbr title="PHPUnit\Framework\ExceptionWrapper">ExceptionWrapper</abbr></a>|null
    <strong>getPreviousWrapped</strong>()
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td><a href="../../PHPUnit/Framework/ExceptionWrapper.html"><abbr title="PHPUnit\Framework\ExceptionWrapper">ExceptionWrapper</abbr></a>|null</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method_setClassName">
        <div class="location">at line 75</div>
        <code>                    void
    <strong>setClassName</strong>(string $className)
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
                            <h4>Parameters</h4>

                    <table class="table table-condensed">
                    <tr>
                <td>string</td>
                <td>$className</td>
                <td></td>
            </tr>
            </table>

            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td>void</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method_setOriginalException">
        <div class="location">at line 80</div>
        <code>                    void
    <strong>setOriginalException</strong>(<a target="_blank" rel="noopener" href="https://www.php.net/Throwable">Throwable</a> $t)
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
                            <h4>Parameters</h4>

                    <table class="table table-condensed">
                    <tr>
                <td><a target="_blank" rel="noopener" href="https://www.php.net/Throwable">Throwable</a></td>
                <td>$t</td>
                <td></td>
            </tr>
            </table>

            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td>void</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
                    <div class="method-item">
                    <h3 id="method_getOriginalException">
        <div class="location">at line 99</div>
        <code>                    <abbr title="PHPUnit\Framework\Throwable">Throwable</abbr>|null
    <strong>getOriginalException</strong>()
        </code>
    </h3>
    <div class="details">
            
    

                <div class="tags">
            
                            <h4>Return Value</h4>

                    <table class="table table-condensed">
        <tr>
            <td><abbr title="PHPUnit\Framework\Throwable">Throwable</abbr>|null</td>
            <td></td>
        </tr>
    </table>

            
            
                    </div>
    </div>

            </div>
            </div>

    
            </div><div id="footer">
        Generated by <a href="https://github.com/code-lts/doctum">Doctum, a API Documentation generator and fork of Sami</a>.</div></div>
    </div>
    </body>

</html>
